### DISCUSSION TITLE
A way to ensure that engine will run at the end of submisson?

### DISCUSSION THREAD

**Discussion by RadekOlenski** [sebas77/Svelto.ECS#85] [Q&A] [answered]
Hey, is there any way to ensure that IReactOnRemoveEx for some component will happen at the very end of submisson? The thing is that I need to dispose some arrays/dictionaries when destroying entities, but if it happens before other remove engines they operates on disposed memory. Or maybe there is a better way to handle Dispose that in remove react engines?

**Comment by sebas77** [comment, marked as answer]
I think what you need is to use the interface IReactOnSubmission. The call back is called once per engine at the end of the whole submission.

**Reply by Grigler** [reply]
It sounds more like he's wanting to order IReactOnRemoveEx engines so that a disposal in one doesn't cause issues in another, though finding myself wanting to order reactions is usually a sign of a design mistake

**Reply by RadekOlenski** [reply]
@Grigler that's true, it is what have crossed my mind. Although using IReactOnSubmission sounds like a much better idea. But actually I have one more question about this: is it possible to query entities that are being removed during the submission in IReactOnSubmission engine? Because in IReactOnRemoveEx I can access desired component from which I need to Dispose things, but I'm not sure how to do this with IReactOnSubmission.

**Reply by sebas77** [reply]
no, you will need to track them separately, however they won't be in the database anymore if deleted so it's though :D

**Reply by RadekOlenski** [reply]
Okeeey, so do I think correctly, that the best solution for handling Dispose would be to have my own solution for this that will be independent from Svelto? One that I can make sure myself that will happen exactly when I need to Dispose?

**Reply by Grigler** [reply]
Depending on what you're disposing, you could always cache it in the `IReactOnRemoveEx` and then dispose it in the `IReactOnSubmission`

**Reply by RadekOlenski** [reply]
> Depending on what you're disposing, you could always cache it in the `IReactOnRemoveEx` and then dispose it in the `IReactOnSubmission`

Thanks, that seems like a good solution! Will try that :)
